### PR TITLE
Improve Mac Install Script 

### PR DESCRIPTION
**Pull request by dianzrong**
TODO:

- [x] Create helper functions
- [x] Install Python
- [x] Install Git
- [x] Install Tesseract OCR
- [x] Test

### CODE REVIEW

**Comment by KrishPatel13** [comment]
@dianzrong 
I have added a TODO for Installation for Tesseract OCR with  the link for reference.

https://github.com/MLDSAI/OpenAdapt/blob/main/install/install_openadapt.sh#L4

It will be great if you implement that for macOS :)

**Comment by dianzrong** [comment]
@KrishPatel13 Will do!

**Comment by KrishPatel13** [comment]
@dianzrong Thank you! 😊 

Also, Here is the link for the PR that added the TODO, just for reference!

https://github.com/MLDSAI/OpenAdapt/pull/243

File: install/install_openadapt.sh [hunk 1]
```diff
@@ -1,25 +1,131 @@
...
+    if [ -d "../OpenAdapt" ]; then
+        cd ..
+        rm -rf OpenAdapt
+        echo "Deleted OpenAdapt directory"
+    fi
+}
+
+# Refresh Path Environment variable
+Refresh() {
+    export PATH="$PATH:$(echo "$PATH" | tr ':' '\n' | grep -v -e '^$' -e '^/usr/local/share/dotnet' -e '^/usr/local/bin' | uniq | tr '\n' ':')$(echo "$PATH" | tr ':' '\n' | grep -e '^/usr/local/share/dotnet' -e '^/usr/local/bin' | uniq | tr '\n' ':')"
```
Comment (abrichr) [inline install/install_openadapt.sh:23]: Can you please try breaking this onto multiple lines? 🙏

File: openadapt/strategies/mixins/svg.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+"""
```
Comment (abrichr) [inline openadapt/strategies/mixins/svg.py:1]: @dianzrong can you please remove this file (and any other unrelated changes that might have made there way in here)? 🙏

File: install/install_openadapt.ps1 [hunk 3]
```diff
@@ -271,4 +271,4 @@ RunAndCheck "alembic upgrade head" "alembic upgrade head"
 RunAndCheck "pytest" "run OpenAdapt tests"
 Write-Host "OpenAdapt installed successfully"
 
-################################   SCRIPT    ################################
+################################   SCRIPT    ################################
```
Comment (KrishPatel13) [inline install/install_openadapt.ps1]: @dianzrong It will be great to remove an extra white space here at Line 274 and push a new commit so that there is no change in ps1

File: install/install_openadapt.sh [hunk 4]
```diff
@@ -31,6 +139,6 @@ pip install -r requirements.txt
 #   [notice] To update, run: pip install --upgrade pip
 pip install -e .
 
-RunAndCheck "alembic upgrade head"
-RunAndCheck "python3.10 -m spacy download en_core_web_trf"
-RunAndCheck "pytest"
\ No newline at end of file
+RunAndCheck "alembic upgrade head" "Update database"
+RunAndCheck "python3.10 -m spacy download en_core_web_trf" "Download spacy model"
```
Comment (KrishPatel13) [inline install/install_openadapt.sh:143]: We can safely remove the installation of trf model for now as it is already included in` poerty install.`

**Comment by KrishPatel13** [comment]
Next up we have to: 

- [ ] Add poetry setup commands instead of .venv

**Review by KrishPatel13 — COMMENTED** [review]
Next up we have to: 

- [x] https://github.com/MLDSAI/OpenAdapt/pull/244#discussion_r1242735949
- [ ] Add poetry setup commands instead of .venv

File: install/install_openadapt.ps1 (hunk 3 above)
Comment (KrishPatel13) [reply install/install_openadapt.ps1]: Resolved in: [42b5e42](https://github.com/MLDSAI/OpenAdapt/pull/244/commits/42b5e42fd8343878a955a9fb81990301f4851843)

File: install/install_openadapt.sh (hunk 4 above)
Comment (dianzrong) [reply install/install_openadapt.sh]: Addressed in [c90084b](https://github.com/MLDSAI/OpenAdapt/pull/244/commits/c90084b140e4b19c52e862da817cc22f3c3acc40)

**Review by KrishPatel13 — COMMENTED** [review]
@dianzrong Can you try to test your script on a machine where it does not have brew, tesseract, python, and git.

This involves using a PC which does not have any of the above aor using Virtual Machine (eg. VMBox) if this works well and create a recording/detailed transcript and putting in the PR ;)

eg. https://github.com/MLDSAI/OpenAdapt/pull/312#issuecomment-1606505563

**Comment by abrichr** [comment]
@dianzrong we tested this on @flyguy712 's machine, and it worked. Thank you!

**Comment by dianzrong** [comment]
For reference: [Video of install](https://drive.google.com/file/d/1yd0cClrh1tG-e5yd20sSo_KEzxPFj9uw/view?usp=sharing)
